packages/kit/src/core/sync/sync.js

```diff
@@ -6,14 +6,24 @@ import { write_matchers } from './write_matchers.js';
 import { write_root } from './write_root.js';
 import { write_tsconfig } from './write_tsconfig.js';
 import { write_types } from './write_types.js';
+import { write_env } from './write_env.js';
 
-/** @param {import('types').ValidatedConfig} config */
-export function init(config) {
+/**
+ * Initialize SvelteKit's generated files.
+ * @param {import('types').ValidatedConfig} config
+ * @param {string} mode
+ */
+export function init(config, mode) {
 	copy_assets(path.join(config.kit.outDir, 'runtime'));
+
 	write_tsconfig(config.kit);
+	write_env(config.kit, mode);
```

Comment:
why would this only be done at `init`? if someone changes an environment variable, don't we want to pick it up?

Reply:
The only way they can change an env var that we can observe is if they edit an `.env` file, in which case Vite restarts (and `init` happens again)